src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -5987,6 +5987,68 @@ private ExpressionAst ErrorRecoveryParameterInExpression(ParameterToken paramTok
                     new CommandParameterAst(paramToken.Extent, paramToken.ParameterName, null, paramToken.Extent)});
         }
 
+        private ExpressionAst ListLiteralRule()
+        {
+            //G  list-literal-expression
+            //G      '['   new-lines:opt   list-entries   new-lines:opt  ']'
+            //G  list-entries
+            //G      unary-expression
+            //G      list-entries   ','   new-lines:opt   unary-expression
+
+            var lBracket = NextLBracket();
+            if (lBracket == null) { return null; }
+
+            var listValues = new List<ExpressionAst>();
+            Token commaToken = null;
+
+            do {
+                if (commaToken != null) { SkipToken(); }
+                V3SkipNewlines();
+
+                var expr = UnaryExpressionRule();
+                if (expr == null)
+                {
+                    // It's OK to have an empty list '[]'
+                    if (commaToken == null) { break; }
+
+                    // ErrorRecovery: create an error expression for the ast and break.
+                    ReportIncompleteInput(After(commaToken), () => ParserStrings.MissingExpressionAfterToken, commaToken.Text);
+                    expr = new ErrorExpressionAst(commaToken.Extent);
+                    listValues.Add(expr);
+                    break;
+                }
+                listValues.Add(expr);
+
+                commaToken = PeekToken();
+            } while (commaToken.Kind == TokenKind.Comma);
+
+            V3SkipNewlines();
+            var rBracket = NextToken();
+
+            if (rBracket.Kind != TokenKind.RBracket)
+            {
+                // ErrorRecovery: Assume only the closing bracket is missing, continue as though it was present.
+                UngetToken(rBracket);
+                ReportIncompleteInput(rBracket.Extent, () => ParserStrings.MissingEndBracketInListExpression);
+            }
+
+            IScriptExtent extent;
+            if (rBracket.Kind == TokenKind.RBracket)
+            {
+                extent = ExtentOf(lBracket, rBracket.Extent);
+            }
+            else if (listValues.Count > 0)
+            {
+                extent = ExtentOf(lBracket, listValues[listValues.Count-1].Extent);
```

Comment:
This excludes a trailing comma. The simple approach to tracking the end is to just have a variable `endExtent` and keep updating it with each token that you consume or ast you add to `listValues`.

Reply:
Update to use a `endExtent` variable.